src/ng-daterangepicker/ng-daterangepicker.component.ts

```diff
@@ -89,15 +90,11 @@ export class NgDateRangePickerComponent implements ControlValueAccessor, OnInit,
   ngOnInit() {
     this.opened = false;
     this.date = dateFns.startOfDay(new Date());
-    this.options = this.options || this.defaultOptions;
+    this.options = _.extend(this.options || {}, this.defaultOptions);
```

Comment:
Doesn't your example create a new object instance? Shouldn't `this.options` when using the component be the same object pointer as the component `this.options` so that value changes such as `this.options.theme` are recognised?